R/detect.r

```diff
@@ -104,3 +104,53 @@ str_ends <- function(string, pattern, negate = FALSE) {
          }
   )
 }
+
+#' Pattern Conversion Between LIKE and Regex structures
+#'
+#' Turns a SQL LIKE structure into a recognisible regular expression pattern.
+#'
+#' @param pattern SQL LIKE operator pattern to be converted.
+#'
+#' @return A converted pattern.
+#'
+#' @examples
+#' like_converter("bana%")
+#' like_converter("app_e")
```

Comment:
You don't need to document this function because it's not exported; and generally I think it's better to put little helper functions like this at the bottom of the file (so you see the big picture first, before the details)

Reply:
And I think `like_to_regex` would be a slightly better name.